### ISSUE TITLE
Add `prefer-tuple` rule

### ISSUE THREAD

**Issue by lo1tuma** [marcalexiei/eslint-plugin-zod#363]
### 🧩 Description

Would you consider a `prefer-tuple` rule for both `eslint-plugin-zod` and `eslint-plugin-zod-mini`?

A length-checked `z.array()` still infers as `T[]`, so the length is lost at the type level. `z.tuple()` encodes it:

```ts
const pair = z.array(z.string()).length(2);
//    ^? z.infer → string[]           // length lost
const pair = z.tuple([z.string(), z.string()]);
//    ^? z.infer → [string, string]   // precise
```

Same for the `mini` API:

```ts
z.array(z.string()).check(z.length(2)); // → z.tuple([z.string(), z.string()])
```

It would fire on a `z.array()` carrying a length constraint:
* `.length()` / `.check(z.length())`
* `.min()` / `.check(z.minLength())`
* `.max()` / `.check(z.maxLength())`

Ideally this should be also autofixable.

FWIW: `z.array(x).nonempty()` already creates correct types `[T, ...T[]]` but there is no equivalent method in zod-mini, zo zod-mini users have to use `.check(z.minLength())`

### 📦 Target plugin(s)

- [x] eslint-plugin-zod
- [x] eslint-plugin-zod-mini

### ⚙️ Options

N/A

### ✅ Valid Code Examples

```typescript
const pair = z.tuple([z.string(), z.string()]);
```

### ❌ Invalid Code Examples

```typescript
const pair = z.array(z.string()).length(2);
```

**Comment by marcalexiei** [comment]
> Would you consider a prefer-tuple rule for both eslint-plugin-zod and eslint-plugin-zod-mini?

Yes, I can add it to both plugins.

**Comment by marcalexiei** [comment]
Available in:
- [`eslint-plugin-zod@4.8.0`](https://github.com/marcalexiei/eslint-zod/releases/tag/eslint-plugin-zod%404.8.0)  
- [`eslint-plugin-zod-mini@1.5.0`](https://github.com/marcalexiei/eslint-zod/releases#release-eslint-plugin-zod-mini@1.5.0).
